### PR TITLE
Feat edit bookmark post

### PR DESCRIPTION
**Pull request by Vishnu-Aithal**
Add Edit and Bookmark Features for Question Answers 

Please Login to check above features

[Live Link](https://neo-support-git-feat-edit-bookmark-post-vishnu-aithal.vercel.app/)

### CODE REVIEW

File: src/components/Profile/BookmarkedAnswers.jsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { Preview } from "../Previews/Preview";
+
+import { useSelector } from "react-redux";
+import { useState } from "react";
+import { useMyBookmarkedAnswers } from "utils/firebase-utils";
+
+export const BookMarkedAnswers = () => {
+    const [bookMarkedAnswers, setBookMarkedBookmarkedAnswers] = useState([]);
```
Comment (ApurvaSawant11) [inline src/components/Profile/BookmarkedAnswers.jsx:8]: Why have you used the word `Bookmarked` twice in the set function? 
Also, try to keep variable/function names as small as possible.

File: src/utils/firebase-utils/answers.js [hunk 2]
```diff
@@ -92,14 +96,67 @@ export const useMyAnswers = (userId, actionCreator) => {
...
+};
+
+export const addBookMarkAnswer = async (post, userId) => {
+    try {
+        const answerRef = doc(db, "answers", post.uid);
+        await updateDoc(answerRef, { bookmarkedBy: arrayUnion(userId) });
+        toast.success("Bookmarked!");
+    } catch (error) {
+        toast.error("Something Went Wrong!");
+        console.log(error);
```
Comment (ApurvaSawant11) [inline src/utils/firebase-utils/answers.js:150]: You can use `console.error()` for showing the error in the console.

**Review by ApurvaSawant11 — APPROVED** [review]
Looks good to me! You can merge the PR

File: src/components/Profile/BookmarkedAnswers.jsx (hunk 1 above)
Comment (Vishnu-Aithal) [reply src/components/Profile/BookmarkedAnswers.jsx:8]: Will Do it . Thanks For Pointing it Out

File: src/components/Posts/PostButtons.jsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                    onClick={() =>
+                        isBookMarked
+                            ? removeBookMarkPost(post, currentUser.uid)
+                            : addBookMarkPost(post, currentUser.uid)
+                    }
+                    className={` rounded-md text-slate-100  hover:scale-105 bg-blue-500 transition-all p-0.5 ${
+                        isBookMarked ? "scale-125 hover:scale-125" : ""
+                    }`}>
+                    {isBookMarked ? (
+                        <CheckIcon className={"w-4 h-4"} />
```
Comment (Dev-AkshatGupta) [inline src/components/Posts/PostButtons.jsx:53]: Why JSX  I don't get it? If not necessary you can remove that

File: src/components/SignIn/InputField.jsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+            <input
+                value={value}
+                onChange={onChange}
+                className="border-b p-1 outline-none bg-transparent placeholder-transparent peer "
+                type={
+                    type !== "password"
+                        ? type
+                        : showPassword
+                        ? "text"
+                        : "password"
```
Comment (Dev-AkshatGupta) [inline src/components/SignIn/InputField.jsx:17]: I liked this

File: src/components/SignIn/SignIn.jsx [hunk 5]
```diff
@@ -1,26 +1,147 @@
...
+                        Sign in with Google
+                    </span>
+                    <GoogleIcon className="h-5 w-5 ml-6 group-hover:scale-110 transition-transform" />
+                </button>
+                <button
+                    onClick={() =>
+                        signInWithEmailPassword({
+                            email: process.env.REACT_APP_TEST_EMAIL,
+                            password: process.env.REACT_APP_TEST_PASSWORD,
+                        })
```
Comment (Dev-AkshatGupta) [inline src/components/SignIn/SignIn.jsx:133]: Really nice use of environment variables

**Review by Dev-AkshatGupta — APPROVED** [review]
Overall great work Vishnu keep on doing amazing work
Yo can merge it just a couple of suggestions been given

File: src/components/Posts/PostButtons.jsx (hunk 3 above)
Comment (Vishnu-Aithal) [reply src/components/Posts/PostButtons.jsx:53]: vs code intellisense 😅

**Pull request merged**